src/Core/Grid/Definition/Factory/CustomerAddressGridDefinitionFactory.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Definition\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\RowActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\Type\LinkRowAction;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\DataColumn;
+
+/**
+ * Class CustomerAddressGridDefinitionFactory defines customer's addresses grid structure.
+ */
+final class CustomerAddressGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    use DeleteActionTrait;
+
+    const GRID_ID = 'customer_address';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return null;
```

Comment:
I don't think the name should be `null`? 🤔

Reply:
I think this was put there in order to avoid printing the grid name (see modifications in grid_panel.html.twig), in order to avoid rendering duplicate "Adresses" string

However I agree the PHP class (which contains the data) should, as much as possible, not be mixed with the template (which defines how it's rendered)

I would rather suggest
- provide an option into Grid to disable the "title" name inside any Grid
- override locally the grid to have this specific grid different from others

The idea being to keep into the View (as in MVC) whatever rendering logic we put, and keep the Grid (the Model) "consistent"